c++/.bazelrc

```diff
@@ -8,6 +8,7 @@ build:unix --cxxopt='-Wno-strict-aliasing' --host_cxxopt='-Wno-strict-aliasing'
 build:unix --cxxopt='-Wno-sign-compare' --host_cxxopt='-Wno-sign-compare'
 build:unix --cxxopt='-Wno-unused-parameter' --host_cxxopt='-Wno-unused-parameter'
 build:unix --cxxopt='-Wno-deprecated-this-capture' --host_cxxopt='-Wno-deprecated-this-capture'
+build:unix --cxxopt='-DCAPNP_HEADER_WARNINGS'
 
```

Comment:
Also add `KJ_HEADER_WARNINGS` (same thing for KJ) and `CAPNP_DEBUG_TYPES` (enabled by default in non-bazel build, code size overhead is acceptable for debug builds) as well, the latter surfaces another warning.